### ISSUE TITLE
Moonlight not using VAAPI/VDPAU with Intel Media SDK installed

### ISSUE THREAD

**Issue by DiegoPomares** [moonlight-stream/moonlight-qt#40]
I get the following error when Moonlight starts up:

_No functioning hardware accelerated H.264 video decoder was detected by Moonlight. Your streaming performance may be severely degraded in this configuration. Click the Help button for more information on solving this problem._

However, I configured VAAPI and VDPAU and they work fine, I tested with a video in VLC and there is a noticeable difference in CPU usage, more info:
```
➜  ~ vainfo                               
libva info: VA-API version 0.39.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_39
libva info: va_openDriver() returns 0
vainfo: VA-API version: 0.39 (libva 1.7.0)
vainfo: Driver version: Intel i965 driver for Intel(R) Skylake - 1.7.0
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Simple            :	VAEntrypointEncSlice
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointEncSlice
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileH264MultiviewHigh      :	VAEntrypointVLD
      VAProfileH264MultiviewHigh      :	VAEntrypointEncSlice
      VAProfileH264StereoHigh         :	VAEntrypointVLD
      VAProfileH264StereoHigh         :	VAEntrypointEncSlice
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD
      VAProfileNone                   :	VAEntrypointVideoProc
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileJPEGBaseline           :	VAEntrypointEncPicture
      VAProfileVP8Version0_3          :	VAEntrypointVLD
      VAProfileVP8Version0_3          :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointEncSlice
```
```
➜  ~ vdpauinfo                                                                                            
display: :0   screen: 0
libva info: VA-API version 0.39.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_39
libva info: va_openDriver() returns 0
API version: 1
Information string: OpenGL/VAAPI/libswscale backend for VDPAU

Video surface:

name   width height types
-------------------------------------------
420     1920  1080  NV12 YV12 UYVY YUYV Y8U8V8A8 V8U8Y8A8 
422     1920  1080  NV12 YV12 UYVY YUYV Y8U8V8A8 V8U8Y8A8 
444     1920  1080  NV12 YV12 UYVY YUYV Y8U8V8A8 V8U8Y8A8 

Decoder capabilities:

name                        level macbs width height
----------------------------------------------------
MPEG1                          --- not supported ---
MPEG2_SIMPLE                   --- not supported ---
MPEG2_MAIN                     --- not supported ---
H264_BASELINE                  51 16384  2048  2048
H264_MAIN                      51 16384  2048  2048
H264_HIGH                      51 16384  2048  2048
VC1_SIMPLE                     --- not supported ---
VC1_MAIN                       --- not supported ---
VC1_ADVANCED                   --- not supported ---
MPEG4_PART2_SP                 --- not supported ---
MPEG4_PART2_ASP                --- not supported ---
DIVX4_QMOBILE                  --- not supported ---
DIVX4_MOBILE                   --- not supported ---
DIVX4_HOME_THEATER             --- not supported ---
DIVX4_HD_1080P                 --- not supported ---
DIVX5_QMOBILE                  --- not supported ---
DIVX5_MOBILE                   --- not supported ---
DIVX5_HOME_THEATER             --- not supported ---
DIVX5_HD_1080P                 --- not supported ---
H264_CONSTRAINED_BASELINE      51 16384  2048  2048
H264_EXTENDED                  --- not supported ---
H264_PROGRESSIVE_HIGH          --- not supported ---
H264_CONSTRAINED_HIGH          --- not supported ---
H264_HIGH_444_PREDICTIVE       --- not supported ---
HEVC_MAIN                      --- not supported ---
HEVC_MAIN_10                   --- not supported ---
HEVC_MAIN_STILL                --- not supported ---
HEVC_MAIN_12                   --- not supported ---
HEVC_MAIN_444                  --- not supported ---

Output surface:

name              width height nat types
----------------------------------------------------
B8G8R8A8         16384 16384    y  
R8G8B8A8         16384 16384    y  
R10G10B10A2      16384 16384    y  
B10G10R10A2      16384 16384    y  
A8               16384 16384    y  

Bitmap surface:

name              width height
------------------------------
B8G8R8A8         16384 16384
R8G8B8A8         16384 16384
R10G10B10A2      16384 16384
B10G10R10A2      16384 16384
A8               16384 16384

Video mixer:

feature name                    sup
------------------------------------
DEINTERLACE_TEMPORAL             -
DEINTERLACE_TEMPORAL_SPATIAL     -
INVERSE_TELECINE                 -
NOISE_REDUCTION                  -
SHARPNESS                        -
LUMA_KEY                         -
HIGH QUALITY SCALING - L1        -
HIGH QUALITY SCALING - L2        -
HIGH QUALITY SCALING - L3        -
HIGH QUALITY SCALING - L4        -
HIGH QUALITY SCALING - L5        -
HIGH QUALITY SCALING - L6        -
HIGH QUALITY SCALING - L7        -
HIGH QUALITY SCALING - L8        -
HIGH QUALITY SCALING - L9        -

parameter name                  sup      min      max
-----------------------------------------------------
VIDEO_SURFACE_WIDTH              -  
VIDEO_SURFACE_HEIGHT             -  
CHROMA_TYPE                      -  
LAYERS                           -  

attribute name                  sup      min      max
-----------------------------------------------------
BACKGROUND_COLOR                 -  
CSC_MATRIX                       -  
NOISE_REDUCTION_LEVEL            -  
SHARPNESS_LEVEL                  -  
LUMA_KEY_MIN_LUMA                -  
LUMA_KEY_MAX_LUMA                -  
```
And this is what I get when running Moonlight from console:
```
➜  ~ flatpak run --branch=stable --arch=x86_64 --command=moonlight com.moonlight_stream.Moonlight
(process:2): Gtk-WARNING **: Locale not supported by C library.
	Using the fallback 'C' locale.
00:00:00 - Qt Warning: Qt: Session management error: None of the authentication protocols specified are supported
00:00:00 - Qt Warning: qrc:/gui/PcView.qml:287:13: QML Rectangle: Binding loop detected for property "implicitWidth"
libva info: VA-API version 0.39.4
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/dri/i965_drv_video.so
libva info: va_openDriver() returns -1
00:00:00 - SDL Error (0): Failed to initialize VAAPI: -1
Failed to open VDPAU backend libvdpau_i965.so: cannot open shared object file: No such file or directory
[AVHWDeviceContext @ 0x1a52d40] VDPAU device creation on X11 display :99.0 failed.
00:00:00 - SDL Error (0): Failed to create VDPAU context: -1313558101
00:00:00 - SDL Info (0): Using SDL software renderer
00:00:00 - SDL Info (0): FFmpeg-based video decoder chosen
00:00:00 - Qt Info: Loaded unique ID from settings: "96e93d6ecb75901e"
00:00:00 - Qt Debug: Current Moonlight version: "0.2.0"
```

My guess is that has something to do with flatpak sandbox environment, maybe I'm missing an extension or some other package?

Full disclosure, first time using flatpak so I'm pretty much lost when it comes to troubleshooting it.

**Comment by DiegoPomares** [comment]
**Recap of our chat in Discord**
The problem was that the driver file in the flatpack sandbox _/usr/lib/dri/i965_drv_video.so_ is a broken symlink pointing to: _/usr/lib/dri/intel-vaapi-driver/i965_drv_video.so_

The actual driver in the sandbox seems to be _/usr/lib/dri/i965_dri.so_, but there is no way to tell VAAPI to use this file instead, since you can only tell VAAPI [what directory the driver is in](https://wiki.archlinux.org/index.php/Hardware_video_acceleration#Configuration). VDPAU is also no help for the same reason.

## Workaround
Copy the VAAPI driver file to a directory into your home (/home/user/dir) and run moonlight as follows:
`flatpak run --env=LIBVA_DRIVERS_PATH=/home/user/dir --filesystem=~/dir:ro --command=moonlight com.moonlight_stream.Moonlight`

**Comment by cgutman** [comment]
Can you try running `flatpak update` and see what packages are updated or installed? I suspect it may be this issue - https://gitlab.gnome.org/GNOME/gnome-software/issues/235

**Comment by DiegoPomares** [comment]
No effect, nothing was updated or installed

**Comment by nQ8hzkZw** [comment]
I managed to get it working by installing VAAPI runtime:
`~ $ flatpak install flathub org.freedesktop.Platform.VAAPI.Intel/x86_64/1.6 -v`

**Comment by danofun** [comment]
This issue appears to be solved in v0.8.1

**Comment by cgutman** [comment]
Awesome! I think the new Flatpak runtime fixed it.
